tests/agent_server/test_resource_sets.py

```diff
@@ -93,3 +93,430 @@ async def test_resource_sets_via_put_version(server, client, environment, client
     resource_sets_from_db = {resource.resource_id: resource.resource_set for resource in resource_list}
     expected_resource_sets = {**resource_sets, "test::Resource[agent1,key=key4]": None}
     assert resource_sets_from_db == expected_resource_sets
+
```

Comment:
I would be nice to have a test that verifies one complex scenario. A test that starts with:
* resources in several different resource sets
* resources in the shared set.

The partial update does:
* An update of a subset of the resources sets
* An addition of shared resources

Reply:
I would even further extend it. Let's also add a new resource set and remove one. I don't think there is already a test that verifies these scenarios, is it?